core/modules/loader/Central.cc

```diff
@@ -0,0 +1,84 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+// Class header
+#include "Central.h"
+
+// system headers
+#include <boost/asio.hpp>
+#include <iostream>
+
+// Third-party headers
+
+
+// qserv headers
+#include "loader/LoaderMsg.h"
+#include "proto/ProtoImporter.h"
+#include "proto/loader.pb.h"
+
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.loader.Central");
+}
+
+namespace lsst {
+namespace qserv {
+namespace loader {
+
+
+Central::~Central() {
+    _loop = false;
+    _pool->shutdownPool();
+    for (std::thread& thd : _ioServiceThreads) {
+        thd.join();
+    }
+}
+
+
+void Central::run() {
+    std::thread thd([this]() { _ioService.run(); });
+    _ioServiceThreads.push_back(std::move(thd));
+}
+
+
+void Central::_checkDoList() {
+    while(_loop) {
+        // Run and then sleep for a second. A more advanced timer should be used
+        _doList.checkList();
+        usleep(100000);
+    }
+}
+
+
+std::ostream& operator<<(std::ostream& os, ChunkSubchunk csc) {
```

Comment:
For the sake of efficiency the second parameters needs to be:
`ChunkSubchunk const& csc`